#Remove hard coded -xSSE4.2 compilation flag for Intel compilers (add the usual -xHOST)
#compilation would fail on non-SSE4.2 CPUs, and for further CPUs/compilers it would restrict the generated code to SSE4.2
# February 19th, 2016 by B. Hajgato - Free University Brussels (VUB)
--- ./numpy/distutils/intelccompiler.py.orig	2016-01-06 22:07:54.000000000 +0100
+++ ./numpy/distutils/intelccompiler.py	2016-02-19 15:07:00.438235792 +0100
@@ -54,7 +54,7 @@
     def __init__(self, verbose=0, dry_run=0, force=0):
         UnixCCompiler.__init__(self, verbose, dry_run, force)
         self.cc_exe = ('icc -m64 -fPIC -fp-model strict -O3 '
-                       '-fomit-frame-pointer -openmp -xSSE4.2')
+                       '-fomit-frame-pointer -openmp -xHOST')
         compiler = self.cc_exe
         if platform.system() == 'Darwin':
             shared_flag = '-Wl,-undefined,dynamic_lookup'
--- ./numpy/distutils/fcompiler/intel.py.orig	2016-01-07 03:16:25.000000000 +0100
+++ ./numpy/distutils/fcompiler/intel.py	2016-02-19 15:07:16.424220745 +0100
@@ -123,7 +123,7 @@
         return ['-openmp -fp-model strict']
 
     def get_flags_arch(self):
-        return ['-xSSE4.2']
+        return ['-xHOST']
 
 # Is there no difference in the version string between the above compilers
 # and the Visual compilers?
